.dotnet/api/OpenAI.netstandard2.0.cs

```diff
@@ -374,7 +374,7 @@ public class FunctionToolDefinition : ToolDefinition, IJsonModel<FunctionToolDef
         public FunctionToolDefinition();
```

Comment:
We need to use the `required` keyword on these if we provide a public default constructor -- that is, if we want to support use like:

```csharp
Tools =
{
    new FunctionToolDefinition()
    {
        FunctionName = "my_function",
    }
},
```

Without the keyword, there's no enforcement of the required properties being present.

I thought we'd agreed in the architecture discussion to keep the `required` property with default constructor support *together* with the constructor that accepts all required properties.

If we want to remove `required`, I'd like to understand why but also don't think we can do it without also removing the public default constructor.

Reply:
I've removed the default constructor as it aligns better with our established guidelines. According to the [guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-model-types), we should provide public constructors with required parameters instead of relying on property initializers, as this ensures that necessary properties are always set.

Currently, only two models use the `required` modifier, which is inconsistent with the rest of our models. I think a better approach is to remove the `required` modifier from these models and provide only public constructors with required parameters. This way, we maintain consistency and follow best practices.